Yoast/YoastSEO.js
# Text 2 tree builder

## Summary

This PR can be summarized in the following changelog entry:

* _[not-user-facing]_ Adds functionality to build a structured tree representation of the source text to analyze.

### Note for testers.
This functionality is not yet used in production. In the (near) future the assessments will be (one-by-one) rewritten to use this tree representation, instead of the current string representation.

## Relevant technical choices:

* Adds the `parse5` library as a basic HTML parser.
* Implements a `TreeAdapter` interface to generate a tree to fit our needs.

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test`, check if the tests pass and the coverage is 100%.
* Review the tests in `buildTreeSpec.js` and the added code used for building the tree. See if you can think of any (edge) cases that are not tested yet and add them as comments to this PR.

Fixes #2028
